### PR TITLE
Bokeh selection colors

### PR DESCRIPTION
**Pull request by philippjfr**
As requested in https://github.com/ioam/holoviews/issues/1139 this adds full control over selection and unselection glyph styles letting you apply custom colors and alphas. Additionally this also adds control of a new feature in bokeh 0.12.5, which is muting glyphs with an interactive legend, this styling can also be controlled via style option but default to a ``alpha=0.2``. This means all our bokeh legends are now clickable.

- [x] Update Bokeh_Backend Tutorial unselection glyph styling examples.
- [x] Tests to check updating all the different glyphs is working correctly.

### CODE REVIEW

**Comment by jlstevens** [comment]
Just curious but what is a ``muted_alpha``?

**Comment by jbednar** [comment]
The amount of alpha to use when something is muted?

**Comment by jlstevens** [comment]
> The amount of alpha to use when something is muted?

Ah, now I get it - maybe``deselected_alpha`` would be a more obvious name, but it is what it is.

**Comment by philippjfr** [comment]
>Ah, now I get it - maybedeselected_alpha would be a more obvious name, but it is what it is.

Right before this I had only added support for ``unselected_alpha/color`` because that had seemed like a more sensible name, but it turned out that was really limiting so now I'm just letting the user set ``selection_color/alpha``, ``unselection_color/alpha`` and ``muted_color/alpha``, which is what bokeh calls the different concepts and provides full control.

**Comment by jbednar** [comment]
Sounds good.  Great stuff!

File: holoviews/plotting/bokeh/chart.py [hunk 1]
```diff
@@ -45,7 +46,7 @@ class PointPlot(LegendPlot, ColorbarPlot):
       Function applied to size values before applying scaling,
       to remove values lower than zero.""")
 
-    style_opts = (['cmap', 'palette', 'marker', 'size', 's', 'alpha', 'color',
+    style_opts = (['cmap', 'palette', 'marker', 'size', 's', 'color',
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/chart.py]: I feel ``style_opts`` should be a set - something for 2.0 perhaps? I am assuming 'alpha' was being duplicated...

File: holoviews/plotting/bokeh/chart.py [hunk 2]
```diff
@@ -170,7 +148,7 @@ class VectorFieldPlot(ColorbarPlot):
        Whether the lengths will be rescaled to take into account the
        smallest non-zero distance between two vectors.""")
 
-    style_opts = ['color'] + line_properties
+    style_opts = line_properties
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/chart.py]: Good to see this simplified.

**Comment by jlstevens** [comment]
Looks good! I'll merge once the tests pass.

**Comment by jlstevens** [comment]
Tests are passing. Merging.

**Pull request merged**